### ISSUE TITLE
Keeping ViewStates?

### ISSUE THREAD

**Issue by PaulWoitaschek** [bluelinelabs/Conductor#174]
I have the following scenario:

I am doing a questionnaire and therefore I have 1 root controller with 5 child controllers I display one after the other each showing a few questions.

When Im in the last controller I display an overview of the previously made answers. When I click on an answer I direct the user back to the controller where the question was asked. 
At that point the controller is still filled with the made answers so the user can edit his answer.
When now going forward I want to display prefilled controllers so no answers are lost.

I'm having a hard time to realize this scenario using conductor. How can this be handled?

**Comment by nomisRev** [comment]
So if I understand correctly you want to remove the backstack of the children you used for the questions. (moving forward) and show the questions in another controller? How about just using Bundle with keys used by every child controller and moving that along the way? When going back you can recover values from the Bundle and when going forward you can recover all the valued you want.

At the end you have an bundle with all the answers.
Correct me if I understood the question wrong.

**Comment by PaulWoitaschek** [comment]
Nope, I go A-B-C-D-E. Then in E the user says: I want to edit something in B. So back to B but after that to the old C, then the old D, then the old E.

**Comment by nomisRev** [comment]
Okay, so so you make all controllers have Bundle constructors)
```java
AController(Bundle bundle)
BController(Bundle bundle)
...
``` 

Now within A, if you want to save your answer add it to the Bundle. And pass that Bundle to BController.

```java
Bundle bundle = getArgs();
bundle.putString("A_ANSWER",answer)
BController bcontroller = BController(bundle)
//push controller
```

Now you have A's bundle in B. If you add the answer of B to the Bundle you got there (again getArgs()) and go back, the answer of A and B will be available in A. When going back to B you can now just recover it from the Bundle.

So basically you're adding all your answer to the same bundle (instance) and passing it along, because you're the same instance the answers will also be available when going back in the backstack.

Hope that helped :+1:

**Comment by PaulWoitaschek** [comment]
Yes that will work. Thanks!
